src/OpenTelemetry.Abstractions/Trace/SpanData.cs

```diff
@@ -20,14 +20,18 @@ namespace OpenTelemetry.Trace
     using System.Collections.Generic;
     using System.Linq;
     using OpenTelemetry.Common;
+    using OpenTelemetry.Resources;
     using OpenTelemetry.Trace.Export;
 
-    /// <inheritdoc/>
-    public sealed class SpanData : ISpanData
+    /// <summary>
+    /// Span data with read-only properties.
+    /// </summary>
+    public sealed class SpanData
```

Comment:
Should be abstract or at least allow to override. Not yet in a spec. But here is a justification pending: https://github.com/open-telemetry/opentelemetry-specification/pull/78/files

Reply:
Abstract leads to errors in other code that is now unable to instantiate an instance.  Removed sealed to allow override.